Makefile

```diff
@@ -0,0 +1,90 @@
+#***************************************************************************
+# set options
+#***************************************************************************
+
+export LATEST_RELEASE_REDIRECT = $(shell curl -s https://github.com/curl/curl/releases/latest -w'%{redirect_url}\n' -o /dev/null)
+
+export LATEST_RELEASE_TAG = $(shell basename $(LATEST_RELEASE_REDIRECT))
+export LATEST_RELEASE_VERSION = $(shell echo $(LATEST_RELEASE_REDIRECT)| cut -d'-' -f 2)
+
+# set curl configure options
+export CONFIGURE_BUILD_OPTS=-"--enable-static --disable-shared --disable-ldap --enable-ipv6 --enable-unix-sockets --with-ssl"
+
+# set docker build options used when building docker images
+export DOCKER_BUILD_OPTS=--compress --squash
+
+# set docker build args used when building docker images
+export DOCKER_BUILD_ARGS =--build-arg CURL_CONFIGURE_OPTION=${configure_build_options} \
```

Comment:
These are too far away from the place they're used. Let's just squash this into one Makefile - there's no need for multiple makefiles right now. I'd rather Docker was being called with these options right next to the `docker` command.

Reply:
I have  left this abstraction in - if only to deal with the inevitable 'can we have this with this os/platform' - I will venture that curl will never have more then 10 images (so no need for buildkit or other abstractions) but potentially more then one and I find managing that complexity better with separate makefiles